src/InputField.js

```diff
@@ -36,29 +36,27 @@ const getInputStyles = showLabel => {
       }
 }
 
-class InputField extends React.Component {
-  constructor(props) {
-    super(props)
-
-    let hasInitialValue
-
-    React.Children.forEach(props.children, child => {
-      if (child && child.type === Input) {
-        hasInitialValue = !!child.props.value
-      }
-    })
+const noop = () => {}
 
-    this.state = {
-      showLabel: hasInitialValue
-    }
+class InputField extends React.Component {
+  static defaultProps = {
+    // for backwards-compatibility
+    onChange: noop
   }
 
-  onInputChange = event => {
-    this.setState({
-      showLabel: event.target.value
-    })
+  // for backwards-compatibility
+  handleChange = onChange => e => {
+    this.props.onChange(e)
+    if (typeof onChange !== 'function') return
```

Comment:
+1 on these `onChange` in a chained functional calls, with proper inner `onChange` type check.